.github/workflows/deployment.yaml

```diff
@@ -1,29 +1,35 @@
-name: Deploy to ECS
+name: Terraform (Plan||Apply)
+run-name: Terraform ${{ inputs.terraform_action }} ${{ inputs.workspace }} by @${{ github.actor }}
 
 on:
   workflow_dispatch:
     inputs:
-      environment:
-        description: 'The environment to deploy to'
+      workspace:
```

Comment:
Non-breaking note: this will be a point of divergence from the Azure repositories, as that side does not leverage TF workspaces as a feature. There should be documentation to explain the discrepancy or choice of architecture somewhere.

Reply:
I do think we should consider a discussion on whether workspaces or discrete folders are preferred. I tend to be wholly against workspaces for some of the reasons that Hashicorp outlines here: https://developer.hashicorp.com/terraform/cli/workspaces#use-cases If the partners you're speaking with overwhelmingly use workspaces, then we shouldn't reinvent the wheel...but, I would be very resistant to using this same approach with Azure.

Reply:
😬 This repo has used workspaces from the beginning; these changes are just around language to eliminate confusion. We should setup a call to discuss.

Reply:
In the examples for this repo, we will continue with workspace usage and encourage clients to utilize separate folders for even more separation.